crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -929,6 +929,119 @@ impl Measured2d for Annulus {
     }
 }
 
+/// A rhombus primitive, also known as a diamond shape.
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[doc(alias = "Diamond")]
+pub struct Rhombus {
+    /// Size of the horizontal and vertical diagonals of the rhombus
+    pub half_diagonals: Vec2,
+}
+impl Primitive2d for Rhombus {}
+
+impl Default for Rhombus {
+    /// Returns the default [`Rhombus`] with a half-horizontal and half-vertical diagonal of `0.5`.
+    fn default() -> Self {
+        Self {
+            half_diagonals: Vec2::splat(0.5),
+        }
+    }
+}
+
+impl Rhombus {
+    /// Create a new `Rhombus` from a vertical and horizontal diagonal sizes.
+    #[inline(always)]
+    pub fn new(horizontal_diagonal: f32, vertical_diagonal: f32) -> Self {
+        Self {
+            half_diagonals: Vec2::new(horizontal_diagonal / 2.0, vertical_diagonal / 2.0),
+        }
+    }
+
+    /// Get the length of each side of the rhombus
+    #[inline(always)]
+    pub fn side(&self) -> f32 {
+        (self.half_diagonals.x.powi(2) + self.half_diagonals.y.powi(2)).sqrt()
+    }
+
+    /// Get the radius of the circumcircle on which all vertices
+    /// of the rhombus lie
+    #[inline(always)]
+    pub fn circumradius(&self) -> f32 {
+        self.half_diagonals.x.max(self.half_diagonals.y)
+    }
+
+    /// Get the radius of the largest circle that can
+    /// be drawn within the rhombus
+    #[inline(always)]
+    #[doc(alias = "apothem")]
+    pub fn inradius(&self) -> f32 {
+        let horizontal_diagonal = self.half_diagonals.x * 2.0;
+        let vertical_diagonal = self.half_diagonals.y * 2.0;
+        (horizontal_diagonal * vertical_diagonal)
+            / (2.0 * (horizontal_diagonal.powi(2) + vertical_diagonal.powi(2)).sqrt())
```

Comment:
```suggestion
        let horizontal_diagonal = self.half_diagonals.x * 2.0;
        let vertical_diagonal = self.half_diagonals.y * 2.0;
        (horizontal_diagonal * vertical_diagonal)
            / (2.0 * (horizontal_diagonal.powi(2) + vertical_diagonal.powi(2)).sqrt())
```
Expanding out the definition for `horizontal|vertical_diagonal`, we arrive at:
```suggestion
        (self.half_diagonals.x * self.half_diagonals.y) / self.side()
```
Geometrical interpretation is that `self.half_diagonals.x * self.half_diagonals.y` is `2*Area` of one quarter right triangle of the rhombus. The inradius of the rhombus is just the height of this triangle opposite to the hypotenuse, hence it's `2*Area / Hypotenuse`.

Also note:
This panics when the rhombus is 0, but the method should probably just return 0 in that case.